src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -93,31 +154,64 @@ public RoadAccident getAccidentByIndex(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+        return roadAccidentList.stream()
+        		.filter(r -> r.getLongitude() > minLongitude).filter(r -> r.getLongitude() < maxLongitude)
+        		.filter(r -> r.getLatitude() > minLatitude).filter(r -> r.getLatitude() < maxLatitude)
+        		.collect(Collectors.toList());
     }
 
     /**
      * find the weather conditions which caused max number of incidents
      * @return
      */
     public List<String> getTopThreeWeatherCondition(){
-        return null;
+		Map<String, Integer> map = new HashMap<String, Integer>();
+		roadAccidentList.stream().map(RoadAccident::getWeatherConditions).forEach(r -> {
```

Comment:
Here not all, that Java 8 could propose. 
use - GroupingBy

map = roadAccidentList.stream().collect(groupingBy( RoadAccident::getWeatherConditions,Collectors.counting()))  - will do the same.